The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/11194810063/job/31121747588) [(plain)](https://github.com/rust-lang/rust/commit/7c8e26bdef2587c39b8f54e5797c4035a4cd55fc/checks/31121747588/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#13 2.930 Building wheels for collected packages: reuse
#13 2.931   Building wheel for reuse (pyproject.toml): started
#13 3.190   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.191   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132715 sha256=dfa09868353292d98f811d3efdb0d54d07389e808efc71d68e3b93c514bf8bec
#13 3.191   Stored in directory: /tmp/pip-ephem-wheel-cache-omk6vfaa/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#13 3.194 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#13 3.603 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#13 3.603 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 4.155 Collecting virtualenv
#13 4.155 Collecting virtualenv
#13 4.195   Downloading virtualenv-20.26.6-py3-none-any.whl (6.0 MB)
#13 4.346      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 6.0/6.0 MB 40.4 MB/s eta 0:00:00
#13 4.402 Collecting platformdirs<5,>=3.9.1
#13 4.406   Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
#13 4.426 Collecting distlib<1,>=0.3.7
#13 4.437      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 468.9/468.9 KB 88.3 MB/s eta 0:00:00
#13 4.473 Collecting filelock<4,>=3.12.2
#13 4.477   Downloading filelock-3.16.1-py3-none-any.whl (16 kB)
#13 4.477   Downloading filelock-3.16.1-py3-none-any.whl (16 kB)
#13 4.559 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 4.751 Successfully installed distlib-0.3.8 filelock-3.16.1 platformdirs-4.3.6 virtualenv-20.26.6
#13 DONE 4.8s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      210880 kB
DirectMap2M:     9226240 kB
DirectMap1G:     9437184 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.05s
##[endgroup]
downloading https://static.rust-lang.org/dist/2024-09-22/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/2024-09-22/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt
---
   Compiling tidy v0.1.0 (/checkout/src/tools/tidy)
    Finished `release` profile [optimized] target(s) in 32.22s
##[endgroup]
fmt check
Diff in /checkout/library/core/src/mem/maybe_uninit.rs:1004:
     /// [`assume_init_mut`]: MaybeUninit::assume_init_mut
     #[unstable(feature = "maybe_uninit_slice", issue = "63569")]
     #[cfg_attr(bootstrap, rustc_allow_const_fn_unstable(const_mut_refs))]
-    #[rustc_const_unstable(
-        feature = "maybe_uninit_slice",
-    )]
-    )]
+    #[rustc_const_unstable(feature = "maybe_uninit_slice", issue = "63569")]
     #[inline(always)]
     pub const unsafe fn slice_assume_init_mut(slice: &mut [Self]) -> &mut [T] {
         // SAFETY: similar to safety notes for `slice_get_ref`, but we have a
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/library/core/src/panicking.rs" "/checkout/library/core/src/range.rs" "/checkout/library/core/src/cell.rs" "/checkout/library/core/src/panic.rs" "/checkout/library/core/src/clone.rs" "/checkout/src/librustdoc/clean/blanket_impl.rs" "/checkout/library/core/src/bool.rs" "/checkout/src/librustdoc/clean/types/tests.rs" "/checkout/library/core/src/panic/unwind_safe.rs" "/checkout/library/core/src/panic/panic_info.rs" "/checkout/library/core/src/panic/location.rs" "/checkout/src/librustdoc/clean/cfg/tests.rs" "/checkout/src/librustdoc/clean/utils/tests.rs" "/checkout/src/librustdoc/clean/simplify.rs" "/checkout/src/librustdoc/clean/inline.rs" "/checkout/src/librustdoc/clean/render_macro_matchers.rs" "/checkout/src/librustdoc/clean/cfg.rs" "/checkout/src/librustdoc/clean/auto_trait.rs" "/checkout/library/core/src/sync/mod.rs" "/checkout/src/librustdoc/visit_ast.rs" "/checkout/library/core/src/sync/exclusive.rs" "/checkout/src/librustdoc/lint.rs" "/checkout/library/core/src/sync/atomic.rs" "/checkout/src/librustdoc/scrape_examples.rs" "/checkout/src/librustdoc/error.rs" "/checkout/library/core/src/borrow.rs" "/checkout/src/librustdoc/fold.rs" "/checkout/library/core/src/tuple.rs" "/checkout/src/librustdoc/externalfiles.rs" "/checkout/library/core/src/lib.miri.rs" "/checkout/src/librustdoc/visit.rs" "/checkout/src/librustdoc/core.rs" "/checkout/src/librustdoc/visit_lib.rs" "/checkout/src/librustdoc/doctest/markdown.rs" "/checkout/src/librustdoc/doctest/rust.rs" "/checkout/src/librustdoc/doctest/tests.rs" "/checkout/src/librustdoc/doctest/runner.rs" "/checkout/src/librustdoc/doctest/make.rs" "/checkout/src/librustdoc/docfs.rs" "/checkout/src/librustdoc/lib.rs" "/checkout/library/core/src/future/join.rs" "/checkout/library/core/src/future/mod.rs" "/checkout/library/core/src/future/into_future.rs" "/checkout/library/core/src/future/async_drop.rs" "/checkout/library/core/src/future/future.rs" "/checkout/library/core/src/future/poll_fn.rs" "/checkout/library/core/src/future/pending.rs" "/checkout/library/core/src/future/ready.rs" "/checkout/library/core/src/cmp.rs" "/checkout/library/core/src/marker.rs" "/checkout/src/librustdoc/json/import_finder.rs" "/checkout/src/librustdoc/json/mod.rs" "/checkout/src/librustdoc/json/conversions.rs" "/checkout/library/core/src/range/legacy.rs" "/checkout/library/core/src/range/iter.rs" "/checkout/library/core/src/pin.rs" "/checkout/library/core/src/prelude/mod.rs" "/checkout/library/core/src/prelude/common.rs" "/checkout/library/core/src/mem/maybe_uninit.rs" "/checkout/library/core/src/mem/mod.rs" "/checkout/library/core/src/mem/manually_drop.rs" "/checkout/library/core/src/mem/transmutability.rs" "/checkout/library/core/src/lib.rs" "/checkout/library/core/src/macros/mod.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Sat Oct  5 16:54:51 UTC 2024
  network time: Sat, 05 Oct 2024 16:54:51 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/core/src/lib.rs
+++ b/library/core/src/lib.rs
@@ -131,7 +131,6 @@
 #![feature(const_ipv6)]
 #![feature(const_likely)]
 #![feature(const_make_ascii)]
-#![feature(const_maybe_uninit_assume_init)]
 #![feature(const_nonnull_new)]
 #![feature(const_num_midpoint)]
 #![feature(const_option)]

--- a/library/core/src/mem/maybe_uninit.rs
+++ b/library/core/src/mem/maybe_uninit.rs
@@ -913,7 +913,11 @@ impl<T> MaybeUninit<T> {
     /// };
     /// ```
     #[stable(feature = "maybe_uninit_ref", since = "1.55.0")]
-    #[rustc_const_unstable(feature = "const_maybe_uninit_assume_init", issue = "none")]
+    #[cfg_attr(bootstrap, rustc_allow_const_fn_unstable(const_mut_refs))]
+    #[rustc_const_stable(
+        feature = "const_maybe_uninit_assume_init",
+        since = "CURRENT_RUSTC_VERSION"
+    )]
     #[inline(always)]
     pub const unsafe fn assume_init_mut(&mut self) -> &mut T {
         // SAFETY: the caller must guarantee that `self` is initialized.
@@ -999,7 +1003,8 @@ impl<T> MaybeUninit<T> {
     ///
     /// [`assume_init_mut`]: MaybeUninit::assume_init_mut
     #[unstable(feature = "maybe_uninit_slice", issue = "63569")]
-    #[rustc_const_unstable(feature = "const_maybe_uninit_assume_init", issue = "none")]
+    #[cfg_attr(bootstrap, rustc_allow_const_fn_unstable(const_mut_refs))]
+    #[rustc_const_unstable(feature = "maybe_uninit_slice", issue = "63569")]
     #[inline(always)]
     pub const unsafe fn slice_assume_init_mut(slice: &mut [Self]) -> &mut [T] {
         // SAFETY: similar to safety notes for `slice_get_ref`, but we have a

